dstep/translator/MacroDefinition.d

```diff
@@ -66,7 +67,92 @@ void resolveMacroDependency(Context context, string spelling)
         return context.includeHandler.resolveDependency(*cursor);
 }
 
-string translate(Expression expression, Context context, Set!string params, ref Set!string imports)
+struct ExpressionContext
+{
+    Context context;
+    Set!string params;
+    Set!string* imports;
+    Cursor[] scope_;
+    alias context this;
+
+    @disable this();
+
+    private this (int x) { }
+
+    static ExpressionContext make()
+    {
+        struct AA { Set!string aa; }
+        auto result = ExpressionContext(0);
+        result.imports = &(new AA()).aa;
+        return result;
+    }
+
+    static ExpressionContext make(Context context)
+    {
+        auto result = ExpressionContext.make();
+        result.context = context;
+        return result;
+    }
+
+    static ExpressionContext make(Context context, Set!string params)
+    {
+        auto result = ExpressionContext.make(context);
+        result.params = params;
+        return result;
+    }
```

Comment:
Is there a reason why constructors are not used?

Reply:
Yes, you cannot create the associative array in the constructor.